contracts/libraries/LibYieldGeneration.sol

```diff
@@ -93,14 +93,13 @@ library LibYieldGeneration {
     /// @notice Conversion from eth to shares
     /// @param _collateralAmount The amount of collateral to withdraw
     /// @param _yield The yield generation struct
-    function _ethToShares(
+    function _neededShares(
         uint _collateralAmount,
         LibYieldGenerationStorage.YieldGeneration storage _yield
     ) internal view returns (uint) {
-        uint pricePerShare = IZaynVaultV2TakaDao(_yield.providerAddresses["ZaynVault"])
-            .getPricePerFullShare();
-
-        return ((_collateralAmount * 10 ** 18) / pricePerShare);
+        return
+            (((_collateralAmount * 10 ** 18) / _yield.totalDeposit) * _yield.totalShares) /
```

Comment:
Could we get rid of the decimal multiplication and divisions by doing:

`_collateralAmount * _yield.totalShares /  _yield.totalDeposit`

Reply:
Commit 6044c03